soc/arm/microchip_mec/mec1501/soc.c

```diff
@@ -92,6 +95,154 @@ static int soc_clk32_init(void)
 	return 0;
 }
 
+/*
+ * Enable peripheral interrupt in ECIA and NVIC.
+ * irq b[7:0] = NVIC external input.
+ *     b[15:8] = GIRQ bit number 0 <= bitnum < 32
+ *     b[23:16] = GIRQ connected to NVIC external input. 8 <= girq_no <= 26
+ *     b[31:24] = unused = 0
+ */
+void z_soc_irq_enable(u32_t irq)
```

Comment:
Originally I looked at riscv32 but the ARM architecture code defined z_arch_xxx and I did not want to touch arch/arm.  Also I need more information than can be encoded in a single byte. The MEC15xx/17xx interrupt routing is so complicated we need 4 bytes. I can't pass 4-bytes to IRQ_CONNECT() without significantly changing it because it uses bytes 1/2 as level 2/3 indicators.  This is like an avalanche, every idea results in having to change more code at the architecture level.